### ISSUE TITLE
"pre instead of div"

### ISSUE THREAD

**Issue by pw374** [ocamllabs/opam-doc#15]
There are things like `<code><code>foo</code>bar<code>plop</code></code>`.
I suspect that `code` can't be nested in `code` because some stuff look awful.

Perhaps we should use span instead of code (with the appropriate css)...

edit: the main suspect has become a `pre`!

**Comment by pw374** [comment]
Arrrrgh... There are `div`s inside `pre`s !

**Comment by pw374** [comment]
I'm not sure yet, but I'm guessing that in `generate.ml`

```
  let h_f =
    match type_decl.typ_manifest, type_decl.typ_kind with
      | None, Ttype_variant _ -> fun x -> Html.pre (Html.code x)
      | None, Ttype_record _ -> fun x -> Html.pre (Html.code x)
      | _ -> Html.pre
  in
```

should be

```
  let h_f =
    match type_decl.typ_manifest, type_decl.typ_kind with
      | None, Ttype_variant _ -> fun x -> Html.pre (Html.code x)
      | None, Ttype_record _ -> fun x -> Html.pre (Html.code x)
      | _ -> Html.div
  in
```

instead.

(It's the most suspicious snippet I could find just by reading. I'll investigate further tomorrow.)

**Comment by pw374** [comment]
The mistakes were not where I thought they were. I am currently fixing them. (I might have finished: I'm testing.)

**Comment by pw374** [comment]
fixed in 97f6d012c547a9c6952364d6bdef6b47a49eb234
